lib/buildFile.js

```diff
@@ -42,8 +45,49 @@ function buildFile(destination, format, platform, dictionary, filter) {
   if (!fs.existsSync(dirname))
     fs.mkdirsSync(dirname);
 
-  fs.writeFileSync(destination, format(filterProperties(dictionary, filter), platform));
-  console.log(chalk.bold.green('✔︎') + ' ' + destination);
+  let filteredProperties = filterProperties(dictionary, filter);
+
+  // Check for property name Collisions
+  var nameCollisionObj = {};
+  filteredProperties.allProperties && filteredProperties.allProperties.forEach((propertyData) => {
+    let propertyName = propertyData.name;
+    if(!nameCollisionObj[propertyName]) {
+      nameCollisionObj[propertyName] = [];
+    }
+    nameCollisionObj[propertyName].push(propertyData);
+  });
+
+  var PROPERTY_NAME_COLLISION_WARNINGS = GroupMessages.GROUP.PropertyNameCollisionWarnings;
+  var WARNING_COLOR = 'darkorange';
+  var HIGHLIGHT_COLOR = 'orangered';
+  var HELP_COLOR = 'orange';
+  Object.keys(nameCollisionObj).map((propertyName) => {
+    if(nameCollisionObj[propertyName].length > 1) {
+      let collisions = nameCollisionObj[propertyName].map(properties => chalk.keyword(HIGHLIGHT_COLOR)(properties.path.join('.')) + '   ' + chalk.keyword(WARNING_COLOR)(properties.value)).join('\n        ');
+      GroupMessages.add(
+        PROPERTY_NAME_COLLISION_WARNINGS,
+        `Output name ${chalk.keyword(HIGHLIGHT_COLOR).bold(propertyName)} was generated by:\n        ${collisions}`
+      );
+    }
+  });
+
+  let propertyNamesCollisionCount = GroupMessages.count(PROPERTY_NAME_COLLISION_WARNINGS);
+
+  fs.writeFileSync(destination, format(filteredProperties, platform));
+  console.log(chalk.bold.green((propertyNamesCollisionCount > 0) ? '⚠️ ' : '✔︎ ') + ' ' + destination);
+
+  if(propertyNamesCollisionCount > 0) {
+    var propertyNamesCollisionWarnings = GroupMessages.flush(PROPERTY_NAME_COLLISION_WARNINGS).join('\n    ');
+    let title = `While building ${chalk.keyword(HIGHLIGHT_COLOR).bold(dest)}, token collisions were found; output may be unexpected.`;
+    let help = chalk.keyword(HELP_COLOR)([
```

Comment:
Maybe use back-ticks here since all you need the array for is to put new line characters in between each line.  I like this message though. You could probably put `title` and `help` into a single string literal to make this cleaner.

Reply:
I've had massive problems with formatting trying to use multiple backtick sections into each other. This was my solution.

I'm going to hold off on this change for now, and when we get everything else to a good place if you have a good suggestion maybe you can do a PR into this one - it was literally the piece that took me the longest.

Reply:
Sounds good, after we merge this I can try to do a cleanup PR on this area.